### PR TITLE
Proctor tool

### PR DESCRIPTION
**Pull request by dittonjs**
this is the main of the proctor tool and is working. There are still features we need to add to this like csv export, sorting and a few others but the core functionality is done.

### CODE REVIEW

File: client/js/proctor/components/exam_distribution/_exam_distribution.jsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    this.getStudents();
+    this.getTestingCenters();
+    this.props.loadAssignedExams(this.props.params.id);
+  }
+
+  getTestingCenters() {
+    const params = {
+      account_id: this.props.testingCentersAccountId,
+    };
+
```
Comment (dittonjs) [inline client/js/proctor/components/exam_distribution/_exam_distribution.jsx]: tkl

File: client/js/proctor/components/exam_distribution/_exam_distribution.jsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    };
+    this.props.reassignExam(assignedExamId, body);
+  }
+
+  studentExamList() {
+    if (!this.props.ready) { return []; }
+
+    if (this.state.sortBy === 'status') {
+      const list = [];
+      _.each(this.props.studentList, (student) => {
```
Comment (dittonjs) [inline client/js/proctor/components/exam_distribution/_exam_distribution.jsx]: maybe lodash this

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    const selectedCenterId = assignedExam ? `${assignedExam.testing_center_id}` : null;
+    this.selector = null;
+    this.state = {
+      selectedCenterId,
+    };
+  }
+
+  componentWillUpdate(nextProps) {
+    if (!this.props.assignedExam && nextProps.assignedExam) {
+      this.setState({ electedCenterId: `${nextProps.assignedExam.testing_center_id}` });
```
Comment (dittonjs) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: oops

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    }
+  }
+
+  getOptions() {
+    return _.map(this.props.testingCenterList, center => ({
+      name: center.name,
+      value: `${center.id}`,
```
Comment (dittonjs) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: lodash toString

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  getStatus() {
+    if (this.props.assignedExam) {
+      return _.capitalize(this.props.assignedExam.status);
+    }
+    return 'Unassigned';
+  }
+
+  assignExam(studentId, centerId) {
```
Comment (dittonjs) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: do this inline or stuff

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    if (assignedExam) {
+      const date = moment(assignedExam.updated_at).format('DD MMM YY H:m');
+      assignObject = (
+        <div>
+          <div style={{ fontSize: '1.1em' }}>{date.toUpperCase()}</div>
+          <div>{assignedExam.instructor_name}</div>
+        </div>
+      );
+    }
+    if (assignedExam && this.state.selectedCenterId !== `${assignedExam.testing_center_id}`) {
```
Comment (dittonjs) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: elseif

File: client/js/proctor/reducers/testing_centers.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import _ from 'lodash';
+
+const defaultState = {
+  testingCenterList: {},
+};
+
+export default function students(state = defaultState, action) {
```
Comment (dittonjs) [inline client/js/proctor/reducers/testing_centers.js]: students.....

File: client/js/proctor/components/exam_distribution/student_assign.jsx [hunk 8]
```diff
@@ -95,34 +90,34 @@ export default class StudentAssign extends React.Component {
...
-    if (assignedExam) {
-      const date = moment(assignedExam.updated_at).format('DD MMM YY H:m');
-      assignObject = (
-        <div>
-          <div style={{ fontSize: '1.1em' }}>{date.toUpperCase()}</div>
-          <div>{assignedExam.instructor_name}</div>
-        </div>
-      );
-    }
     if (assignedExam && this.state.selectedCenterId !== `${assignedExam.testing_center_id}`) {
```
Comment (BenjaminCall) [inline client/js/proctor/components/exam_distribution/student_assign.jsx]: everywhere else you _.toString it. you want to leave concat there?

**Comment by BenHSmith13** [comment]
Looks good

LGTM

**Pull request merged**